### PR TITLE
Implements tabs horizontal menu(design system)

### LINKED ISSUE #246: [RLD-24] - Tabs - Horizontal Menu
## **Why?**

We are in the process of creating our new core components using [mantine](https://mantine.dev/core/tabs/) and creating our very own [storybook](https://storybook.js.org/docs/react/get-started/introduction) for easier UI development.

To learn more about our design system see [#219](https://github.com/notifirehq/notifire/issues/219)

## **How?**

To start contributing you need to run storybook locally:

```
npm run setup:project
cd apps/web 
npm run storybook
```

To contribute, add a new component directory named `Tabs` in `apps/web/src/design-system/` with the following files:

1\. Component file, overriding Mantine’s default implementation based on the design below.

2\. Styles implementation, to fit our design. Consider dark and light color scheme.

3\. Create a story for the tabs component, showcasing our designs options.

### **Tabs designs**

Tabs dark scheme:

[Frame 237 - Relayed Design System (Figma)](https://www.figma.com/file/KprcTdnkZz0bxFepI8ZSHP/Relayed-Design-System?node-id=595%3A1758)

Tabs light scheme:

[Frame 237 - Relayed Design System (Figma)](https://www.figma.com/file/KprcTdnkZz0bxFepI8ZSHP/Relayed-Design-System?node-id=595%3A1827)

> See the [Button](https://github.com/notifirehq/notifire/tree/master/apps/web/src/design-system/button) or [Switch](https://github.com/notifirehq/notifire/tree/master/apps/web/src/design-system/switch) component for implementation examples.

## **Need help?**

For any questions feel free to contact us on [discord](https://discord.gg/vh6N694qMa) 😃

### PR DESCRIPTION
**Pull request by Abobos**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
 feature

- **What is the current behavior?** (You can also link to an open issue here)
closes #246 

- **What is the new behavior (if this is a feature change)?**
Adds a tabs horizontal menu component/stories to notifire web application design system

- **Other information**:

### CODE REVIEW

File: apps/web/src/design-system/tabs/tabs.styles.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import { createStyles, MantineTheme } from '@mantine/core';
```
Comment (ainouzgali) [inline apps/web/src/design-system/tabs/tabs.styles.ts:1]: Please notice that between two tabs should be 30px, currently, there are 50.
I think adding: 
 ```
   tabsList: {
      gap: '30px',
    }, 
```
should do the trick

**Review by ainouzgali — COMMENTED** [review]
@Abobos It looks great! I really liked your implementation. 
Thank you so much for contributing, it is much appreciated🚀
Just please see some of the notes about the design :)

File: apps/web/src/design-system/tabs/tabs.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  MenuTab: IMenuButtonProp[];
+  orientation?: 'horizontal' | 'vertical';
+  position?: 'right' | 'center' | 'left' | 'apart';
+  withIcon?: boolean;
+}
+
+export function Tab({ MenuTab, orientation = 'horizontal', withIcon = false, position = 'left' }: ITabsProp) {
+  const { classes } = useStyles(withIcon);
+
+  const [activeTab, setActiveTab] = useState(1);
```
[the remark this answers is not in the document]
Comment (Abobos) [reply apps/web/src/design-system/tabs/tabs.tsx:22]: @ainouzgali yes we can leave the state management to the user of this component as this actually makes the component a presentational one. But, how do we now show the `active state` look and feel of the tab menu buttons?

File: apps/web/src/design-system/tabs/tabs.styles.ts (hunk 1 above)
Comment (Abobos) [reply apps/web/src/design-system/tabs/tabs.styles.ts:1]: Noted

File: apps/web/src/design-system/tabs/tabs.styles.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        },
+      },
+    },
+
+    tabLabel: {
+      ref: tabLabel,
+
+      fontSize: '14px',
+      fontWeight: '800',
+      color: theme.colors.gray[5],
```
[the remark this answers is not in the document]
Comment (Abobos) [reply apps/web/src/design-system/tabs/tabs.styles.ts:57]: Noted

File: apps/web/src/design-system/tabs/tabs.styles.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  const tabIcon = getRef('tabIcon');
+
+  return {
+    tabControl: {
+      marginBottom: withIcon ? '30px' : '0',
+    },
+
+    tabActive: {
+      [`& .${tabLabel}`]: {
+        color: dark ? '#FFFFFF' : theme.colors.gray[7],
```
[the remark this answers is not in the document]
Comment (Abobos) [reply apps/web/src/design-system/tabs/tabs.styles.ts:16]: This is cool 💯

**Comment by Abobos** [comment]
@ainouzgali Feedbacks noted. This will be implemented ASAP

**Review by ainouzgali — COMMENTED** [review]
Could you please make the requested changes? and let me know what you think about the active state? 
and then we will merge💃🏼😎

**Comment by Abobos** [comment]
@ainouzgali All reviews have been implemented :)

**Review by ainouzgali — APPROVED** [review]
Everything looks amazing!! 
Had a great time working with you! 🤗

**Pull request merged**